scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/view/grid/jobs/JobsListGrid.java

```diff
@@ -488,17 +487,58 @@ protected void buildCellContextualMenu(Menu menu) {
         removeItem.addClickHandler(event -> controller.removeJob(ids));
         removeItem.setEnabled(selFinished);
 
-        menu.setItems(pauseItem,
-                      restartInErrorTaskItem,
-                      resumeItem,
-                      resumeAndRestartItemTask,
-                      priorityItem,
-                      killItem,
-                      killAndResubmitItem,
-                      resubmitItem,
-                      openItem,
-                      exportXmlItem,
-                      removeItem);
+        SchedulerServiceAsync scheduler = Scheduler.getSchedulerService();
+        boolean finalSelPauseOrRunning = selPauseOrRunning;
+        scheduler.getJobInfoDetails(LoginModel.getInstance().getSessionId(), ids.get(0), new AsyncCallback<String>() {
+            public void onSuccess(String result) {
+                Set<String> signals = parseSignals(result);
+                MenuItem actionsItem = new MenuItem("Actions",
+                                                    SchedulerImages.instance.job_kill_16().getSafeUri().asString());
+                Menu signalsMenu = new Menu();
+                for (final String signal : signals) {
+                    MenuItem item = new MenuItem(signal.substring(signal.indexOf("_") + 1));
+                    item.addClickHandler(event2 -> controller.addJobSignal(signal.substring(signal.indexOf("_") + 1),
+                                                                           ids.get(0)));
+                    signalsMenu.addItem(item);
+                }
+                actionsItem.setSubmenu(signalsMenu);
+                actionsItem.setEnabled(selSingleSelected && finalSelPauseOrRunning && !signals.isEmpty());
+
+                menu.setItems(pauseItem,
+                              restartInErrorTaskItem,
+                              resumeItem,
+                              resumeAndRestartItemTask,
+                              priorityItem,
+                              killItem,
+                              killAndResubmitItem,
+                              resubmitItem,
+                              openItem,
+                              exportXmlItem,
+                              removeItem,
+                              actionsItem);
+
+            }
+
+            public void onFailure(Throwable caught) {
+                String message = JSONUtils.getJsonErrorMessage(caught);
+                LogModel.getInstance().logCriticalMessage("Failed to get job details : " + message);
```

Comment:
why critical ? 
What happens when a job has 0 actions ?

Reply:
When the job has 0 actions, we will receive an empty list and the menu "Actions" will be disabled

Reply:
good, that what I was expecting. Thx